docs/source/developer/translation.rst

```diff
@@ -0,0 +1,128 @@
+Translation
+##############################################################################
+
+
+Translation Principles
+=============================================================================
+
+The following principles were used to guide the design of a translation system for CS Field Guide:
+
+1. Translations should be 'first class citizens' - they should exist as independent entities from the source content, and be coupled as loosely as possible to it.
+2. Translated content should remain accessible even if the source material changes, and an updated translation is not yet available.
+3. Users should be able to see all content that exists, even if it is not available in their language.
+4. Translated content should be presented solely in the users language, without being interspersed with English.
+
+
+Translatable Files
+=============================================================================
+Translatable content is stored in five types of files:
+
+- Markdown files, for content to be processed by Verto.
+- YAML files containg field translations for a given model type.
+- HTML templates.
+- Python code.
+- JavaScript code.
+
+The first two types of file are stored inside the directory tree for a given language (ie. the directory named using the Django locale code).
+
+.. note::
+
+  YAML translation files must contain translations for only one model type, but may contain translations for multiple instances of that model, and for multiple fields.
+
+  They are structured as follows
+
+  .. code-block:: yaml
+
+    <object-slug-1>:
+        <field-1>: <translated value for field-1>
+        <field-2>: <translated value for field-2>
+    <object-slug-2>
+
+  For example, the following snippet is from the YAML translation file for all classroom resources
+
+  .. code-block:: yaml
+
+    pens:
+      description:  Pens
+    paper:
+      description:  Paper
+    number-line-0-20:
+      description:  Number line from 0 to 20
+
+  It is important to note that these YAML files are separate from `configuration files <understanding_configuration_files>`_, which are located in the ``structure`` directory.
+
+  These files can be parsed and loaded using a `utility function <UtilityFunctions_>`_ on the ``TranslatableModelLoader`` base class.
+
+For the other three type of files, Django's built-in `translation support <https://docs.djangoproject.com/en/2.0/topics/i18n/>`_ is utilised to handle translatable strings.
+In Python code, text is wrapped in a ``ugettext`` function call (usually aliased to ``_``).
+In HTML templates, text is wrapped in ``{% trans %}``/``{% blocktrans trimmed %}`` tags.
```

Comment:
For (un)interactives It's almost always {% blocktrans %} rather than {% blocktrans trimmed %} in the html templates